tools/autograd/derivatives.yaml

```diff
@@ -2313,5 +2313,14 @@
 - name: _pin_memory(Tensor self, Device? device=None) -> Tensor
   self: grad
 
+# empty factory functions have explicit non-differentiability so that they propagate the class
+# when used with a Tensor subclass together with __torch_dispatch__.
+# All the other factory functions depend on these, so everything works as expected
```

Comment:
nit: "everything works as expected" is a bit too broad.
Maybe All the other factory functions are composite and call into one of these.